src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.Windows.cs

```diff
@@ -888,5 +889,7 @@ private static string GetEnvironmentVariablesBlock(IDictionary<string, string> s
             // an extra null at the end that indicates end of list will come from the string.
             return stringBuff.ToString();
         }
+
+        private static string GetErrorMessage(int error) => Interop.Kernel32.GetMessage(error);
```

Comment:
the logic for building and formatting the exception message is now present in 3 files: `Process.Windows.cs`, `Process.Win32.cs` and `Process.Unix.cs`. I think that you could create a new private method in `Process.cs` and just reuse it. Sth like this:


```cs
private static Win32Exception CreateDetailedException(string prefix, int errorCode, string fileName, string? workingDirectory)
{
    string directoryForException = string.IsNullOrEmpty(workingDirectory) ? Directory.GetCurrentDirectory() : workingDirectory;
    string msg = SR.Format(SR.FailedToStartFileDirectory, fileName, directoryForException, prefix);
    return new Win32Exception(errorCode, msg);
}
```

Reply:
You are totally right. Will place that code in a common method.